File: src/main/java/com/cloudogu/gitops/infrastructure/git/providers/scmmanager/api/ScmManagerUser.java
package com.cloudogu.gitops.infrastructure.git.providers.scmmanager.api;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashMap;
import java.util.Map;

/**
 * Request payload describing an SCM-Manager user account, created via {@link UsersApi}.
 */
@Getter
@Setter
@NoArgsConstructor
public class ScmManagerUser {
	private String name;
	private String displayName;
	private String mail;
	private boolean external;
	private String password;
	private boolean active = true;

	@JsonProperty("_links")
	private Map<String, Object> links = new HashMap<>();
}

File: src/main/kotlin/com/cloudogu/gitops/infrastructure/git/providers/scmmanager/api/ScmManagerUser.kt
package com.cloudogu.gitops.infrastructure.git.providers.scmmanager.api

import com.fasterxml.jackson.annotation.JsonProperty

class ScmManagerUser {
    var name: String? = null
    var displayName: String? = null
    var mail: String? = null
    var external: Boolean = false
    var password: String? = null
    var active: Boolean = true
    
    @JsonProperty("_links")
    var _links: Map<String, Any> = emptyMap()

    // Standard no-arg constructor
    constructor()

    // Map-constructor for Groovy @CompileStatic map-coercion
    constructor(map: Map<String, Any?>) {
        (map["name"] as? String)?.let { this.name = it }
        (map["displayName"] as? String)?.let { this.displayName = it }
        (map["mail"] as? String)?.let { this.mail = it }
        (map["external"] as? Boolean)?.let { this.external = it }
        (map["password"] as? String)?.let { this.password = it }
        (map["active"] as? Boolean)?.let { this.active = it }
        (map["_links"] as? Map<*, *>)?.let { 
            @Suppress("UNCHECKED_CAST")
            this._links = it as Map<String, Any> 
        }
    }
}
